### ISSUE TITLE
Docker nomad driver volumes and network_mode

### ISSUE THREAD

**Issue by orenmazor** [hashicorp/waypoint#505]
I currently run several nomad services that talk to each other. They both run with network_mode=host, and share a volume. 

I'm playing around with getting these deployed via waypoint, but I'm not seeing any way to do this yet. Is it possible and I'm just missing it, or is it just early days? (which is obviously totally cool)

**Comment by mitchellh** [comment]
The set of attributes we support with the Nomad plugin is fairly small right now. What is missing for you to get going? We can add them.

**Comment by orenmazor** [comment]
That totally makes sense. this is day 1 of release :)

I think just the volume and network settings are in my way

**Comment by mitchellh** [comment]
Could you show us a quick pseudo-example of what you're looking for?

**Comment by orenmazor** [comment]
sure! this is a rough nomad job hcl of what I'd like (and this in fact is running already in a small prototype). the goal is to have two tasks running in one group. they communicate with each other via shared volume. one also exposes itself via http.

```
        job "my-job" {
          datacenters = ["dc1"]
          type = "service"
          group "my-tasks" {
            volume "my-volume" {
                type      = "host"
                source    = "my-volume"
            }
            network {
              mode = "host"
              port "http" {}
            }

            task "one" {
              driver = "docker"
              env {
                  OP_PEER="http://127.0.0.1:8081"
              }

              volume_mount {
                  volume      = "my-volume"
                  destination = "/home/special"
              }
              config {
                network_mode = "host"
                image = "ecr-url"
                ports = ["http"]

              }
            }

            task "two" {
              driver = "docker"

              env {
                  OP_PEER="http://127.0.0.1:8080"
                  USE_HTTP_PORT=8081
              }
              volume_mount {
                  volume      = "my-volume"
                  destination = "/home/special"
              }
              config {
                network_mode = "host"
                image = "another-ecr-url"
              }

            }
          }
        }

```

as for a signature of what I'd imagine, even something as simple as this would probably solve my problem, but I'm open to whatever syntax makes the most sense. PS thanks for responding. I feel like there's SOMETHING else going on today that would take up your attention already ;)

```
deploy {
    use "nomad" {
      region = "global"
      datacenter = "dc1"
      network_mode = "host"
       volume_mount {
                  volume      = "my-volume"
                  destination = "/home/special"
       }
      static_environment = {
        "environment": "production",
        "LOG_LEVEL": "debug"
      }
      service_port = 3000
      replicas = 1
    }
}
```

**Comment by orenmazor** [comment]
another thing that would be useful, but lower on my priority is the ability to specify periodic tasks.

**Comment by 42wim** [comment]
> The set of attributes we support with the Nomad plugin is fairly small right now. What is missing for you to get going? We can add them.

Sorry to hijack, this issue. 
I can open another issue if needed. I wonder why there's no cpu/memory option for nomad (but there is for some of the other platforms). Is there a reason for this? Am I missing something about how waypoint works with nomad?

**Comment by orenmazor** [comment]
heyo! just following up to see if this is on any roadmaps?

I'm now deploying this to ~8 different environments. makefile -> waypoint (via deploy exec) -> levant -> nomad.

I'd like to simplify this, to at least not have levant in the pipeline. less moving parts :)

**Comment by koesbong** [comment]
Hey @orenmazor, thanks for the follow up. There is a feature (#1235) we are working on now for 0.4 that will solve this issue. We will close this once that’s been released.

**Comment by mitchellh** [comment]
Fixed by #1299
